drivers/virtio/net/virtio_net.c

```diff
@@ -922,11 +937,17 @@ static int virtio_netdev_feature_negotiate(struct uk_netdev *n)
 		VIRTIO_FEATURE_SET(drv_features, VIRTIO_NET_F_GUEST_CSUM);
 	}
 
+	/* VirtIO modern */
+	if (VIRTIO_FEATURE_HAS(host_features, VIRTIO_F_VERSION_1)) {
+		VIRTIO_FEATURE_SET(drv_features, VIRTIO_F_VERSION_1);
+		vndev->modern = 1;
```

Comment:
The only reason I added the `modern` member field was for determining the header size of virtio-net (either 10 or 12 depending on the version). Now that I think about it, we should store the header size instead:
```
vndev->hdr_size = sizeof(virtio_net_hdr_mrg_rxbuf); 
```

Reply:
I think the `modern` field will be useful for other features later, that are not available in legacy. Also see the comment below on the `is_modern` parameter.